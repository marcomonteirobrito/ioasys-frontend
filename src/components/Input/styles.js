import styled from 'styled-components';

import Colors from '../../utils/Colors';

export const InputValue = styled.input`
  border: none;
  box-shadow: none;
  background: none;

  color: ${Colors.white};

  ::placeholder {
    color: ${Colors.white};
    opacity: 0.5;
    size: 12px;
    line-height: 16px;
  }
`;

export const Label = styled.div`
  color: ${Colors.white};
  opacity: 0.5;
  size: 12px;
  line-height: 16px;
  margin: 8px 0px 4px 16px;
`;

export const InputField = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: flex-start;
`;

export const Button = styled.button`
  width: 85px;
  height: 36px;
  margin: 12px;
  background: ${Colors.white};
  color: ${Colors.buttonLabelColor};
  border-radius: 44px;
  border: none;

  :hover {
    opacity: 0.9;
  }
`;

export const Container = styled.div`
  display: flex;
  justify-content: flex-start;
  align-items: center;
  border-radius: 4px;

  padding: 0 15px;
  width: 368px;
  height: 60px;
  margin-bottom: 16px;

  background: ${Colors.black32};

  &:hover {
    ${InputValue} {
      display: flex;
    }
  }
`;
