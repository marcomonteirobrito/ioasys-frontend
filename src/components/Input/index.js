import React from 'react';
import PropTypes from 'prop-types';

import { InputField, Label, InputValue } from './styles';

function Input({ inputType }) {
  return (
    <InputField>
      <Label>teste</Label>
      <InputValue type={inputType} />
    </InputField>
  );
}

Input.propTypes = {
  inputType: PropTypes.string.isRequired,
};

export default Input;
