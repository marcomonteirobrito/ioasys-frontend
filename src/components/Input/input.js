/* eslint-disable no-unused-vars */
import React from 'react';
import PropTypes from 'prop-types';

import Button from '../Button/button';

import { Container, Label, InputField, InputContainer } from './styles';

/**
 * @param {string} id - Identificação unica do elemento
 * @param {string} name - Nome do elemento
 * @param {string} className - ClassName do elemento
 * @param {string} type - Tipo do input
 * @param {string} inputTitle - Titulo do input
 * @param {string} placeholder - Texto a ser mostrado quando o elemento não tem valor
 * @param {array} value - Valor do elemento
 * @param {event} onChange - Função chamada quando o valor é alterado
 * @param {event} buttonClick - Função chamada quando clicar no botão
 * @param {boolean} button - Se verdadeiro, insere um botão no input
 * @param {boolean} loading - Habilita o loading
 */

const Input = ({
  id,
  name,
  className,
  type,
  inputTitle,
  onChange,
  value,
  placeholder,
  button,
  buttonClick,
  loading,
  disabledButton,
}) => (
  <Container className={className}>
    <InputContainer>
      <Label>{inputTitle}</Label>
      <InputField
        id={id}
        name={name}
        className={className}
        type={type}
        onChange={onChange}
        value={value}
        placeholder={placeholder}
      />
    </InputContainer>

    {button && (
      <Button onClick={buttonClick} loading={loading} disabled={disabledButton}>
        Entrar
      </Button>
    )}
  </Container>
);

Input.propTypes = {
  type: PropTypes.string,
  inputTitle: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
  button: PropTypes.bool.isRequired,
  placeholder: PropTypes.string,
  buttonClick: PropTypes.func,
  name: PropTypes.string,
  className: PropTypes.string,
  id: PropTypes.string,
  loading: PropTypes.bool,
  disabledButton: PropTypes.bool,
};

Input.defaultProps = {
  name: 'input',
  className: 'input-container',
  id: 'input',
  type: 'text',
  placeholder: '',
  buttonClick: () => {},
  loading: false,
  disabledButton: false,
};

export default Input;
