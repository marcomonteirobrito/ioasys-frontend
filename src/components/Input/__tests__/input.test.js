/* eslint-disable no-undef */
import React from 'react';
import { render, screen } from '@testing-library/react';

import Input from '../input';

const renderInput = () =>
  render(
    <Input
      type="text"
      inputTitle="Email"
      onChange={jest.fn()}
      value="marco_monteiro@live.com"
      placeholder="Insira seu email"
      button={false}
      buttonClick={jest.fn()}
      loading={false}
      disabledButton
    />
  );

describe('Input', () => {
  it('Deve renderizar o componente corretamente', () => {
    renderInput();

    expect(document.body).toMatchSnapshot();
  });

  it('NÃO deve renderizar o botão', () => {
    renderInput();

    const button = screen.queryByTestId('button');

    expect(button).toBeNull();
  });

  it('Deve renderizar o botão', () => {
    render(
      <Input
        type="text"
        inputTitle="Email"
        onChange={jest.fn()}
        value="marco_monteiro@live.com"
        placeholder="Insira seu email"
        button={false}
        buttonClick={jest.fn()}
        loading
        disabledButton
      />
    );

    const button = screen.queryByTestId('button');

    expect(button).toBeDefined();
  });
});
