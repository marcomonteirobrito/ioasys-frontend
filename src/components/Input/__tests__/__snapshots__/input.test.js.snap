// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Deve renderizar o componente corretamente 1`] = `
<body>
  <div>
    <div
      class="sc-jSgupP jdcqbq input-container"
    >
      <div
        class="sc-hKgILt eoXUuG"
      >
        <label
          class="sc-dlfnbm dyjhlw"
        >
          Email
        </label>
        <input
          class="sc-gsTCUz dhFEmg input-container"
          id="input"
          name="input"
          placeholder="Insira seu email"
          type="text"
          value="marco_monteiro@live.com"
        />
      </div>
    </div>
  </div>
</body>
`;
